lib/galaxy/model/metadata.py

```diff
@@ -667,7 +667,6 @@ def from_external_value(self, value, parent, path_rewriter=None):
                 # directory. Correct.
                 file_name = path_rewriter(file_name)
             mf.update_from_file(file_name)
-            os.unlink(file_name)
```

Comment:
This isn't the right place to do this, cleaning up the job working directory will handle this correctly. fixes the directory metadata strategy case

Reply:
I bet this is also what happened in https://github.com/galaxyproject/galaxy/pull/14087